README.md

```diff
@@ -435,6 +355,399 @@ One App can be started via docker or if built from source by running `node lib/s
+security and bundle size considerations.
+
+**Contents**
+* [provideStateConfig](#providestateconfig)
+* [csp](#csp)
+* [corsOrigins](#corsorigins)
+* [configureRequestLog](#configurerequestlog)
+* [extendSafeRequestRestrictedAttributes](#extendsaferequestrestrictedattributes)
+* [createSsrFetch](#createssrfetch)
+* [validateStateConfig](#validatestateconfig)
+* [requiredSafeRequestRestrictedAttributes](#requiredsaferequestrestrictedattributes)
+* [appCompatibility](#appcompatibility)
+
+##### `provideStateConfig`
+**Module Type**
+* ✅ Root Module 
+* 🚫 Child Module  
+
+**Shape**
+```js
+if (!global.BROWSER) {
+  Module.appConfig = {
+    provideStateConfig: {
+      server: {
+        [settingName]: {
+          [environmentLevel]: String,
+        },
+      },
+      client: {
+        [settingName]: {
+          [environmentLevel]: String,
+        },
+      },
+    },
+  };
+}
+```
+
+The `provideStateConfig` directive is useful for supplying string-based key value settings per runtime (e.g. `client` or `server`) and per `environmentLevel` (e.g. QA, Prod, etc). The `environmentLevel` is specified in the `ONE_CONFIG_ENV` environment variable when running the Server.
+
+In practice, the state config supplied by a Root Module may look like this shape:
+
+```js
+if (!global.BROWSER) {
+  Module.appConfig = {
+    provideStateConfig: {
+      server: {
+        myApiHostname: {
+          development: 'qa.api.intranet.example.com',
+          qa: 'qa.api.intranet.example.com',
+          production: 'prod.api.intranet.example.com',
+        },
+      },
+      client: {
+        myApiHostname: {
+          development: 'qa.api.external.example.com',
+          qa: 'qa.api.external.example.com',
+          production: 'prod.api.external.example.com',
+        },
+      },
+    },
+  };
+}
+```
+
+Based on `environmentLevel`, the String values are injected into the global [`config` reducer](./src/universal/ducks/config.js) in One App's global Redux state. These values may be accessed by Modules using Redux's `mapStateToProps`.
+
+**📘 More Information**
+* Example: [Frank Lloyd Root's `appConfig`](./prod-sample/sample-modules/frank-lloyd-root/0.0.0/src/config.js)
+* [`validateStateConfig`](#validatestateconfig)
+* Source: [`config` reducer](./src/universal/ducks/config.js)
+
+##### `csp`
+**Module Type**
+* ✅ Root Module 
+* 🚫 Child Module  
+
+⚠️ Required Directive
+
+**Shape**
+```js
+if (!global.BROWSER) {
+  RootModule.appConfig = {
+    csp: String,
+  };
+}
+```
+
+The `csp` static `String` should be a valid [Content Security Policy (CSP)](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP) for your application which will be passed on to the HTML markup rendered by the Browser.
+
+> We recommend using something like [content-security-policy-builder](https://www.npmjs.com/package/content-security-policy-builder) to create your CSP string.
+
+> 👮**Security Feature**: Limits the scripts and assets allowed to load.
```

Comment:
what about grouping this with `⚠️ Required Directive`

Reply:
Fixed in https://github.com/americanexpress/one-app/pull/6/commits/425b78c86d6303601316e35204272ce4bc67956e